### ISSUE TITLE
GitHub page build failure (missing _include symlink) when specifying a `source` directory

### ISSUE THREAD

**Issue by mnyrop** [jekyll/jekyll#7247]
- [x] I believe this to be a bug, not a question about using Jekyll.
- [x] I updated to the latest Jekyll (or) if on GitHub Pages to the latest `github-pages`
- [x] I ran `jekyll doctor` to check my configuration
- [ ] I read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/
- [ ] This is a feature request.

---

- [x] I am on (or have tested on) ***macOS*** 10+
- [x] I am on (or have tested on) ***Debian/Ubuntu*** GNU/Linux
- [ ] I am on (or have tested on) ***Fedora*** GNU/Linux
- [ ] I am on (or have tested on) ***Arch*** GNU/Linux
- [ ] I am on (or have tested on) ***Other*** GNU/Linux
- [ ] I am on (or have tested on) ***Windows*** 10+

---

- [ ] I was trying to install.
- [ ] There is a broken Plugin API.
- [x] I had an error on GitHub Pages, and I have reproduced it locally.
- [ ] I had an error on GitHub Pages, and GitHub Support said it was a Jekyll Bug.
- [ ] I had an error on GitHub Pages and I did not test it locally.
- [ ] I was trying to build.
- [ ] It was another bug.

## My Reproduction Steps

- Build the jekyll site using the `source` directory `src` (works locally; builds without errors and passes tests)
- Push to origin master to have GitHub build the site with GhPages (page build failure, symlink missing)
- Try pushing only the compiled `_site` to `gh-pages` branch via rake task (page build failure, symlink missing)

__Result:__
```
Liquid Exception: Could not locate the included file 'banner.html' in any of 
["/home/travis/build/mnyrop/wax/_includes"]. Ensure it exists in one of those directories and,
if it is a symlink, does not point outside your site source. 
in src/about.md
```
(See line #672 in build [Travis-CI log](https://travis-ci.org/mnyrop/wax/jobs/428276311#L672).)

Basically, gh-pages is not acknowledging my `source` directory and is looking for `_includes` in my jekyll root directory. (In this case, `wax/_includes` instead of `wax/src/_includes`).

Repository: <https://github.com/mnyrop/wax>

## The Output I Wanted
Successful build on GitHub Pages.

**Comment by mnyrop** [comment]
Might be related to [#7224](https://github.com/jekyll/jekyll/pull/7224)?

**Comment by mnyrop** [comment]
Still failing on newest Jekyll release (3.8.4)

**Comment by pathawks** [comment]
GitHub Pages overrides `source`.
https://github.com/github/pages-gem/blob/master/lib/github-pages/configuration.rb#L38-L40

This is not a Jekyll bug, but a restriction of the GitHub Pages environment.

**Comment by mnyrop** [comment]
@pathawks Thanks for clarifying
